### PR TITLE
Test Coverage For Get Fast Healthy Meals Use Case

### PR DESCRIPTION
**Pull request by Belal-Alkhatib**
### Refactoring:
- Replaced `PREPARATION_TIME_TARGET `with clearly named `MIN_PREPARATION_TIME `and `MAX_PREPARATION_TIME `constants.
- Updated `getValidMeals `to ensure preparationTime is strictly within the **1–15** minute range.
- Removed redundant validation from filterHealthyMeals.

### Testing:
Added thorough unit tests to cover:
- Empty and null inputs
- Meals with invalid or borderline nutrition values
- Edge cases for preparation time (including negative and boundary values)
- Ensured correct filtering of healthy and fast meals

### CODE REVIEW

File: src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        getFastHealthyMealsUseCase = GetFastHealthyMealsUseCase(mealsRepository)
+    }
+
+    @Test
+    fun `getFastHealthMeals should throw empty data exception when repository given empty list`() {
+        //Given
+        every { mealsRepository.getAllMeals() } returns emptyList()
+
+        //When & Then
+        assertThrows<FoodMoodException.Validation.EmptyDataException> { getFastHealthyMealsUseCase.getFastHealthMeals() }
```
Comment (MohammadAshrf) [inline src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt]: You can import `FoodMoodException.Validation` up there to simplify the sentence to be like:

```
assertThrows<EmptyDataException> { 
getFastHealthyMealsUseCase.getFastHealthMeals() 
}
```

File: src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+import logic.exception.FoodMoodException
+import logic.helpers.createMeal
+import logic.models.Nutrition
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class GetFastHealthyMealsUseCaseTest {
+    private lateinit var mealsRepository: MealsRepository
+    private lateinit var getFastHealthyMealsUseCase: GetFastHealthyMealsUseCase
```
Comment (MohammadAshrf) [inline src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt]: You should make these variables simpler to keep the reference down there in every test case, also simple.
So I suggest changing the naming here to be just: 

`private lateinit var repository: MealsRepository` 
`private lateinit var useCase: GetFastHealthyMealsUseCase`

File: src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+    @BeforeEach
+    fun setup() {
+        mealsRepository = mockk(relaxed = true)
+        getFastHealthyMealsUseCase = GetFastHealthyMealsUseCase(mealsRepository)
+    }
+
+    @Test
+    fun `getFastHealthMeals should throw empty data exception when repository given empty list`() {
```
Comment (MohammadAshrf) [inline src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt]: You should delete the `getFastHealthMeals` word from all test cases and start with "should" because it's already the same name as the current class, and because we all stick to the same naming convention for functions as a team.

so it will be like that:

```
 @Test
    fun `should throw empty data exception when repository given empty list`() {}
```

**Review by MohammadAshrf — CHANGES_REQUESTED** [review]
Nice work overall Belal

File: src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        every { repository.getAllMeals() } returns mealsOutsidePrepTimeTarget
+
+        //When & Then
+        assertThat(useCase.getFastHealthMeals()).containsExactly(healthyMealResult)
+
+    }
+
+    @Test
+    fun `should return the correct meals when there is a meal with a preparation time exactly at the edge of the preparation time range`() {
+        // Given
```
Comment (mohamedd-hassan) [inline src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt:158]: i feel like the naming of the test here could be improved a bit, does the correct meal include the meals at the edge or are they excluded

File: src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+        every { repository.getAllMeals() } returns listOf(healthyMeal1, healthyMeal2)
+
+        // When & Then
+        assertThat(useCase.getFastHealthMeals()).containsExactly(healthyMeal1)
+    }
+
+    @Test
+    fun `should return the correct meals when all meals are healthy, but it has a preparation time out of range (1 to 15)`() {
+        //Given
```
Comment (mohamedd-hassan) [inline src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt:136]: same thing the name could be a bit clearer, maybe something like:

should return the correct meals when all meals are healthy excluding meals with preparation time out of range (1 to 15)

**Review by mohamedd-hassan — CHANGES_REQUESTED** [review]
very good work, just the minor name changes to clear up the test case to the user

File: src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt (hunk 5 above)
Comment (omer1998) [reply src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt:136]: from the name it seems you're testing two parameters, if it is healthy and if the preparation time out of a specific range, it might need some name changing

Comment (Belal-Alkhatib) [reply src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt:136]: > from the name it seems you're testing two parameters, if it is healthy and if the preparation time out of a specific range, it might need some name changing

you are right guys, I will make that change 👍

File: src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt (hunk 4 above)
Comment (Belal-Alkhatib) [reply src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt:158]: > i feel like the naming of the test here could be improved a bit, does the correct meal include the meals at the edge or are they excluded

good note, i will make that change

**Review by mohamedd-hassan — APPROVED** [review]

File: src/main/kotlin/logic/usecase/GetFastHealthyMealsUseCase.kt [hunk 6]
```diff
@@ -26,10 +31,9 @@ class GetFastHealthyMealsUseCase(
```
Comment (omer1998) [inline src/main/kotlin/logic/usecase/GetFastHealthyMealsUseCase.kt:31]: can you use map for one time and return list contain the average of each needed parameter, it will be a performance gain to loop across all meals just one time

File: src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+        //When & Then
+        assertThrows<EmptyDataException> { useCase.getFastHealthMeals() }
+    }
+
+    @Test
+    fun `should throw empty data exception when all meals have null nutrition`() {
+        //Given
+        every { repository.getAllMeals() } returns listOf(
+            createMeal(nutrition = defaultNutrition, preparationTime = PREPARATION_TIME_ABOVE_TARGET),
+        )
```
Comment (omer1998) [inline src/test/kotlin/logic/usecase/GetFastHealthyMealsUseCaseTest.kt:45]: it might be better to assign nutrition as null as you said in the test case name

**Review by omer1998 — APPROVED** [review]
great work belal
